pennylane/interfaces/execution.py

```diff
@@ -508,8 +544,12 @@ def cost_fn(params, x):
             return_tuple=False,
             pass_kwargs=new_device_interface,
         )
-        results = cached_execute_fn(tapes, execution_config=config)
-        return batch_fn(results)
+        if not transform_program.is_informative():
+            results = cached_execute_fn(tapes, execution_config=config)
+        else:
+            results = tapes
+
+        return _post_processing(results, processing_fns, classical_cotransforms)

```

Comment:
```suggestion
        if transform_program.is_informative():
            return _post_processing(tapes, processing_fns, classical_cotransform)
        return _post_processing(results, processing_fns, classical_cotransforms)
```

This will allow us to:

1) Control on the positive, rather than on a negation. Any negation `not val` forces the reader to store an additional piece of information in their active memory while reading. Any extra piece of information that needs to be in active memory will make the code harder to read.

2) Avoid mixing up possible types for the `results` variable.  When I see a variable named `results`, I expect it to hold numeric results. The violation of the expectation will lead to future devlopers doing things that are not valid operations on the type.